sds_data_manager/stacks/sds_data_manager_stack.py

```diff
@@ -0,0 +1,216 @@
+# Standard
+import pathlib
+
+# Installed
+import aws_cdk as cdk
+from constructs import Construct
+from aws_cdk import (
+    Stack,
+    Environment,
+    RemovalPolicy,
+    aws_iam as iam,
+    aws_s3 as s3,
+    aws_s3_deployment as s3_deploy,
+    aws_secretsmanager as secrets,
+    aws_lambda as lambda_,
+    aws_lambda_event_sources,
+    aws_lambda_python_alpha as lambda_alpha_
+)
+
+# Local
+from .opensearch_stack import OpenSearch
+
+
+class SdsDataManager(Stack):
+    """Stack for Data Management."""
+    def __init__(self, scope: Construct,
+                 construct_id: str,
+                 sds_id: str,
+                 opensearch: OpenSearch,
+                 env: Environment,
+                 **kwargs) -> None:
+        """SdsDataManagerStack
+
+        Parameters
+        ----------
+        scope : App
+        construct_id : str
+        sds_id: str
+        env : Environment
+            Account and region
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        # This is the S3 bucket used by upload_api_lambda
+        data_bucket = s3.Bucket(
+            self, f"DataBucket-{sds_id}",
+            bucket_name=f"sds-data-{sds_id}",
+            versioned=True,
+            removal_policy=RemovalPolicy.DESTROY,
+            auto_delete_objects=True,
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+        )
+
+        # Confirm that a config.json file exists in the expected
+        # location before S3 upload
+        if (
+            not pathlib.Path(__file__)
+            .parent.joinpath("..", "config", "config.json")
+            .resolve()
+            .exists()
+        ):
+            raise RuntimeError(
+                "sds_data_manager/config directory must contain config.json"
+            )
+
+        # S3 bucket where the configurations will be stored
+        config_bucket = s3.Bucket(
+            self,
+            f"ConfigBucket-{sds_id}",
+            bucket_name=f"sds-config-bucket-{sds_id}",
+            versioned=True,
+            removal_policy=RemovalPolicy.DESTROY,
+            auto_delete_objects=True,
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+        )
+
+        # Upload all files in the config directory to the S3 config bucket.
+        # This directory should contain a config.json file that will
+        # be used for indexing files into the data bucket.
+        s3_deploy.BucketDeployment(
```

Comment:
Would it make sense to make the S3 deployment a separate file as well? Genuinely asking, I'm not sure if there are advantages beyond reusing the stacks.

Reply:
@maxinelasp that would make sense. Also how we are doing it here makes sense too. I actually tried that quickly which resulted in a circular dependency (the stacks weren't deploying in the order they were supposed to). And that could have opened a can of worms. Since it didn't really seem to matter either way I left them in.

Reply:
Sounds good! Thanks for checking that. Those circular dependencies are a real pain!